be/src/olap/rowset/segment_v2/column_reader.h

```diff
@@ -91,41 +92,35 @@ class ColumnReader {
 
     bool has_zone_map() const { return _meta.has_zone_map_page(); }
 
-    // get row ranges with zone map
-    // - cond_column is user's query predicate
-    // - delete_conditions is a vector of delete predicate of different version
-    Status get_row_ranges_by_zone_map(CondColumn* cond_column,
-                                      const std::vector<CondColumn*>& delete_conditions,
-                                      OlapReaderStatistics* stats,
-                                      RowRanges* row_ranges);
-
     PagePointer get_dict_page_pointer() const { return _meta.dict_page(); }
 
-private:
-    ColumnReader(const ColumnReaderOptions& opts,
-                 const ColumnMetaPB& meta,
-                 uint64_t num_rows,
-                 RandomAccessFile* file);
-    Status init();
+    ColumnZoneMap* column_zone_map() {
```

Comment:
I don't like the idea to make the internal member public in order to implement `get_row_ranges_by_zone_map` in FileColumnIterator, this breaks encapsulation. I think you can achieve the same goal by adding an output parameter `std::vector<uint8_t>* _page_delete_status` to `get_row_ranges_by_zone_map`.

Reply:
ok